src/develop/pixelpipe_hb.c

```diff
@@ -344,7 +344,13 @@ void dt_dev_pixelpipe_synch(dt_dev_pixelpipe_t *pipe, dt_develop_t *dev, GList *
     piece = (dt_dev_pixelpipe_iop_t *)nodes->data;
     if(piece->module == hist->module)
     {
-      piece->enabled = hist->enabled;
+      const gboolean mustbe_enabled = piece->module->default_enabled && piece->module->hide_enable_button;
+      piece->enabled = hist->enabled || mustbe_enabled;
+      if((hist->enabled == 0) && mustbe_enabled)
+      {
+        fprintf(stderr,"[dt_dev_pixelpipe_synch] alway-on module `%s' found as disabled in history\n", piece->module->op);
```

Comment:
Something to rework here... This is wrong for jpeg, I got this when opening one:

`[dt_dev_pixelpipe_synch] alway-on module `demosaic' found as disabled in history`

For demosaic the `mustbe_enabled` variable must take into account the image format otherwise the message is misleading and acticating the demosaic for non raw image could certainly have bad side effect! I completely missed that when reviewing the code, sorry!

Reply:
Yeah. I also missed that. But i don't think this is just about demosaicing, it could be also in other modules. What we need would be a test for a **module must_be_disabled**. ATM i don't know how to test for that as we take the information from the gtk widget state and that is not reliable.

Reply:
So should we revert this for now ? I don't have a solution for this myself.

Reply:
I think so. Better crashing on a very rare bug than introducing a common new one :-) 

And at least @elstoc has proposed a pr fixing the en-disabling bug via a shortcut.

Unfortunately @chhil could not confirm a keyboard shortcut so we still don't know the trigger for the reported issue.

ATM i am at heavy work with @heckflosse optimising the rcd demosaicer so i don't have much spare time to implement some housekeeping in the modules to do a proper fix and i am also not sure whether it's really worth it.

Reply:
I already confirmed in the main thread. I don't have a shortcut.

https://github.com/darktable-org/darktable/pull/7015#issuecomment-732803932

Reply:
Ok, I'm reverting for now.